src/controllers/hawkerCentreController.ts

```diff
@@ -1,6 +1,45 @@
 import HawkerCentre from '../models/HawkerCentre';
```

Comment:
This looks better than the previous one :) A better way to do it will be have a `preprocess` and `postprocess` middleware right at the front and back of the call stack for all of the routes exported from the controller.

`preprocess` - looks at the request. If there is a need to change from `hawkerCentreId` to `id`, do it.
`postprocess` - looks at the response. If there is a need to change response from `id` to `hawkerCentreId`, do it.

The `getIdFormatting` and `getMultipleIdFormatting` can be merged into one function, so that people using it will not need to care if their response is in an array or a single item. 

If this way of doing is followed, all of the controllers will just set the response code using `res.status(...)` and set the `res.local.body` to the desired response. The `postprocess` middleware will then do a `res.json(res.local.body)` to send it out.

Reply:
Oh so u mean like the postprocessing will handle both sending the response back out and there is no need for the showStall route anymore?

Reply:
Still need `showStall` route as we need to bind `/stalls/:id` to that specific route. Yeap, `postprocess` can actually handle both conversion and sending the response back, after all we have to do conversion on almost every route in the controller.

Reply:
i actually meant the showStall function which returns the response but got it thanks!